lua/telescope/finders.lua

```diff
@@ -15,6 +15,40 @@ local _callable_obj = function()
 end
 
 
+local IncrementalFinder = _callable_obj()
+
+function IncrementalFinder:new(opts)
+  opts = opts or {}
+  local obj = setmetatable({
+    entry_maker = opts.entry_maker or make_entry.from_string,
+  }, self)
+
+  self.results = {}
+  self.finished = false
+
+  obj._find = coroutine.wrap(function(finder, _, process_result, process_complete)
+    repeat
+      for _, result in ipairs(self.results) do
+        process_result(finder.entry_maker(result))
+      end
```

Comment:
Thinking about this. This should not work. Like on each repeat run you will process all results again, not only the newly added once. What about doing something like:
```lua
while table.getn(self.results) > 0 do
  -- pop element and do process_result
  -- But depending on how table is implemented this could be a bad perf hit.
done
```